source/core/StringUtilities.cpp

```diff
@@ -118,25 +119,41 @@ Boolean SubString::CompareCStr(ConstCStr begin2) const
     // The source string cannot have a null before it hit end
     // So if a null is found in the supplied string first the test bails out false.
     while(sCompare < sEnd) {
-        if (*sCompare++ != *pBegin2++) {
+        if (*sCompare++ != *pBegin2++)
             return false;
-        }
     }
     // Comparison is true if supplied string is the same length.
     return (*pBegin2 == 0);
 }
 //------------------------------------------------------------
 Boolean SubString::ComparePrefix(const Utf8Char* begin2, Int32 length2) const
 {
-    if (length2 > Length()) {
+    if (length2 > Length())
```

Comment:
Can we make Compare and ComparePrefix to share code or are there performance implications? Also we have ComparePrefixIgnoreCase, while Compare used to take in a boolean for ignore case, and that is not the case for ComparePrefix.